lib/src/rebuild.rs

```diff
@@ -25,35 +49,37 @@ pub async fn copy(only: Vec<&str>, delay: Duration, copy: bool) -> Result<()> {
     Ok(())
 }
 
-fn copy_to(instance: &Instance, _attempts: u64, copy: bool) -> Result<()> {
-    env::set_var("IP", instance.public_ip.clone());
+fn copy_to(instance: &BitteNode, _attempts: u64, cache: &Option<String>) -> Result<()> {
+    env::set_var("IP", instance.pub_ip.to_string());
     let flake = ".";
 
     handle_command_error(execute::command_args!(
         "nix",
         "run",
         format!(
             "{}#nixosConfigurations.{}.config.secrets.generateScript",
-            flake, instance.uid
+            flake, instance.nixos
         )
     ))?;
 
-    let target = format!("{}#{}", flake, instance.flake_attr);
-    let cache = format!(
-        "{}&secret-key=secrets/nix-secret-key-file",
-        instance.s3_cache
+    let target = format!(
+        "{}#nixosConfigurations.{}.config.system.build.toplevel",
+        flake, instance.nixos
     );
-    let rebuild_flake: String = format!("{}#{}", flake, instance.uid);
+    let rebuild_flake: String = format!("{}#{}", flake, instance.nixos);
 
     nix_build(&target)?;
-    if copy {
+
+    if let Some(c) = cache {
+        let cache = format!("{}&secret-key=secrets/nix-secret-key-file", c);
```

Comment:
I'd love to fetch this value from some file in the encrypted folder instead so we can slowly get rid of the secrets dir.